proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -1150,23 +1150,23 @@ private Block pushBlock(Block block) {
     return block;
   }
 
-  public SymbolScope getCurrentScope() {
+  public TreeParserSymbolScope getCurrentScope() {
     return currentScope;
   }
 
-  public SymbolScopeRoot getRootScope() {
+  public TreeParserRootSymbolScope getRootScope() {
     return rootScope;
   }
 
   private void classStateInherits(JPNode classNode, JPNode inheritsTypeNode) {
     LOG.trace("Entering classStateInherits {} {}", classNode, inheritsTypeNode);
     String className = inheritsTypeNode.attrGetS(IConstants.QUALIFIED_CLASS_STRING);
-    SymbolScopeSuper cachedCopy = null;
+    TreeParserSuperSymbolScope cachedCopy = null;
     try {
-      cachedCopy = SymbolScopeSuper.cache.get(className.toLowerCase(), new Callable<SymbolScopeSuper>() {
+      cachedCopy = TreeParserSuperSymbolScope.cache.get(className.toLowerCase(), new Callable<TreeParserSuperSymbolScope>() {
         @Override
-        public SymbolScopeSuper call() throws Exception {
-          SymbolScopeSuper clz = classStateSuper(classNode, className);
+        public TreeParserSuperSymbolScope call() throws Exception {
+          TreeParserSuperSymbolScope clz = classStateSuper(classNode, className);
           if (clz == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2583)